### DISCUSSION TITLE
Betaflight Target Conversion to INAV  - Script?

### DISCUSSION THREAD

**Discussion by FraserJB** [iNavFlight/inav#10357] [General]
Hi, has anyone successfully made use of this script?
https://github.com/iNavFlight/inav/blob/mosca-target-converter/src/utils/bf2inav.py

I'm attemping to create a custom target for HGLRCF722AIO_X6, I started from the INAV HGLRCF722 which should be very simiar, but whatever I do I can only seem to get 6 out of 5 motors to spin-up (eveything else seems to work okay!)
[note: I can succesfully compile modified target files into the HEX files].

I thought I'd try this script, but is just throwing a "no such file" error.

I'm also open to any other ideas...?

**Comment by mmosca** [comment]
Yes. But it is far from complete and you still need to assign the timers, based on the options provided.

**Reply by FraserJB** [reply]
I managed to get the script to run and output the target and config files. I'll work through it and see if between the INAV HGLRCF722 and the converted HGLRCF722AIO_X6 if I can get something working.

The script is a great idea by the way!

**Comment by lydianb79** [comment]
How do you get around the "no such file" output. How do you run this? I tried with python3 bf2inav.py config.h

**Reply by sensei-hacker** [reply]
> How do you get around the "no such file" output. How do you run this? I tried with python3 bf2inav.py config.h

"No such file" means you gave it the name/location of a file that doesn't exist.  If you're on Windows, Windows may have renamed the file config.h.txt or similar.
You can set your File Explorer options to show file extensions, or use "dir" in the command line to double check the actual name of the file.

**Reply by lydianb79** [reply]
Thanks a lot, I tried on OSX ad on CentOS, same output

iMac:blub blub$ ls -l
total 80
-rw-r--r--@  1 blub  staff   2963 13 Jan 16:24 BEFH-BETAFPVG473.config
-rwxr-xr-x@  1 blub  staff  31822 13 Jan 15:24 bf2inav.py
-rwxr-xr-x@  1 blub  staff   3005 13 Jan 15:33 config.h
drwxrwxr-x@ 28 blub  staff    896 13 Jan 16:55 inav-mosca-target-converter

iMac:blub blub$ python3 bf2inav.py
no such file 

iMac:blub blub$ python3 bf2inav.py config.h
no such file

**Reply by sensei-hacker** [reply]
I see the script expects options for input and output:

mkdir BEFH-BETAFPVG473
python bf2inav.py -i BEFH-BETAFPVG473.config -o  BEFH-BETAFPVG473

**Reply by lydianb79** [reply]
Again thanks a lot for the Support!
Works but the MCU Definition is missing, looks like I have another Puzzle :-)
